### DISCUSSION TITLE
Is it possible to support the use of Redis in the Functions?

### DISCUSSION THREAD

**Discussion by cddqssc** [appwrite/appwrite#6210] [Q&A]
Functions provide a high level of scalability and extensibility. And Many functionalities require caching. 

Is it possible to use Redis caching in Functions?

**Comment by gewenyu99** [comment]
@cddqssc this is a great question. Maybe contribute a cache-enabled container to Open Runtimes? Idk if this is something you might wanna try ;)

**Reply by cddqssc** [reply]
I have carefully read the execution method of cloud functions, and it seems that each execution of a cloud function is completely independent and cannot integrate the state from the previous invocation. 
This means that if I use Redis caching, I would need to reconnect to the Redis server every time it is executed, which is not the behavior I expected. Currently, I have to run the functionality that requires caching on a separate server. 

Thank you for your response.

**Reply by gewenyu99** [reply]
@cddqssc Yes, you might need to contribute a dedicated OPR container that:
1. Has a Redis instance attached. The executor is, in fact, persistent until it is shut down after inactivity.
2. Has a shutdown behavior, maybe save snapshots of the container before it is shutdown? The same image is used until redeployed. So it could be possible to persist the cache.

Just an idea, you'd need to contribute both an OPR container and probably changes to Appwrite.
